src/Text/Parsing/Parser/String.purs

```diff
@@ -2,89 +2,113 @@
+
+-- | Match end of stream.
+eof :: forall f c m. StreamLike f c => Monad m => ParserT f m Unit
 eof = do
   input <- gets \(ParseState input _ _) -> input
-  unless (null input) (fail "Expected EOF")
+  case uncons input of
+    Nothing -> pure unit
+    _ -> fail "Expected EOF"
 
--- | Match the specified string.
-string :: forall s m. StringLike s => Monad m => String -> ParserT s m String
-string str = do
+-- | Match the specified prefix.
+prefix :: forall f c m. StreamLike f c => Show f => Monad m => f -> ParserT f m f
+prefix str = do
   input <- gets \(ParseState input _ _) -> input
-  case indexOf (wrap str) input of
-    Just 0 -> do
+  case stipPrefix (Prefix str) input of
+    Just {rest, updatePos} -> do
       modify \(ParseState _ position _) ->
-        ParseState (drop (length str) input)
-                   (updatePosString position str)
-                   true
+        ParseState rest (updatePos position) true
       pure str
     _ -> fail ("Expected " <> show str)
 
--- | Match any character.
-anyChar :: forall s m. StringLike s => Monad m => ParserT s m Char
-anyChar = do
+-- | Match any token.
+token :: forall f c m. StreamLike f c => Monad m => ParserT f m c
+token = do
   input <- gets \(ParseState input _ _) -> input
   case uncons input of
     Nothing -> fail "Unexpected EOF"
-    Just { head, tail } -> do
+    Just ({ head, updatePos, tail }) -> do
       modify \(ParseState _ position _) ->
-        ParseState tail
-                   (updatePosString position (singleton head))
-                   true
+        ParseState tail (updatePos position) true
       pure head
 
--- | Match a character satisfying the specified predicate.
-satisfy :: forall s m. StringLike s => Monad m => (Char -> Boolean) -> ParserT s m Char
+-- | Match a token satisfying the specified predicate.
+satisfy :: forall f c m. StreamLike f c => Show c => Monad m => (c -> Boolean) -> ParserT f m c
 satisfy f = try do
-  c <- anyChar
+  c <- token
   if f c then pure c
-         else fail $ "Character '" <> singleton c <> "' did not satisfy predicate"
+         else fail $ "Character " <> show c <> " did not satisfy predicate"
+
+-- | Match the specified token
+match :: forall f c m. StreamLike f c => Eq c => Show c => Monad m => c -> ParserT f m c
+match c = satisfy (_ == c) <?> show c
 
--- | Match the specified character
-char :: forall s m. StringLike s => Monad m => Char -> ParserT s m Char
-char c = satisfy (_ == c) <?> show c
 
--- | Match a whitespace character.
-whiteSpace :: forall s m. StringLike s => Monad m => ParserT s m String
-whiteSpace = do
-  cs <- many $ satisfy \c -> c == '\n' || c == '\r' || c == ' ' || c == '\t'
-  pure $ fromCharArray cs
+-- | Match a whitespace characters but returns them using Array.
+whiteSpace :: forall f m. StreamLike f Char => Monad m => ParserT f m (Array Char)
+whiteSpace = many $ satisfy \c -> c == '\n' || c == '\r' || c == ' ' || c == '\t'
```

Comment:
whitespace is frequently used parser and I think it's better to use function here performance wise, instead of `oneof`. if you still want to use `oneof` i can change it.